forc-plugins/forc-fmt/src/main.rs

```diff
@@ -69,29 +77,177 @@ fn run() -> Result<()> {
+    if let Ok(file_content) = fs::read_to_string(&file) {
+        let mut edited = false;
+        let file_content: Arc<str> = Arc::from(file_content);
+        let build_config = manifest_file.map(|f| {
+            BuildConfig::root_from_file_name_and_manifest_path(
+                file.clone(),
+                f,
+                BuildTarget::default(),
+            )
+        });
+        match Formatter::format(formatter, file_content.clone(), build_config.as_ref()) {
             Ok(formatted_content) => {
-                format_file(path, &formatted_content)?;
+                if app.check {
+                    if *file_content != formatted_content {
+                        info!("\n{:?}\n", file);
+                        display_file_diff(&file_content, &formatted_content)?;
+                        edited = true;
+                    }
+                } else {
+                    write_file_formatted(&file, &formatted_content)?;
+                }
+
+                return Ok(edited);
             }
             Err(err) => {
+                // there could still be Sway files that are not part of the build
+                error!("\nThis file: {:?} is not part of the build", file);
                 error!("{}\n", err);
             }
         }
     }
 
-    Ok(())
+    bail!("Could not read file")
+}
+
+/// Format the workspace at the given directory.
+fn format_workspace_at_dir(app: &App, workspace: &WorkspaceManifestFile, dir: &Path) -> Result<()> {
+    let mut contains_edits = false;
+    let mut formatter = Formatter::from_dir(dir)?;
+    let mut members = vec![];
+
+    for member_path in workspace.member_paths()? {
+        members.push(member_path)
+    }
+
+    // Format files at the root - we do not want to start calling format_pkg_at_dir() here,
+    // since this would mean we format twice on each subdirectory.
+    if let Ok(read_dir) = fs::read_dir(dir) {
+        for entry in read_dir.filter_map(|res| res.ok()) {
+            let path = entry.path();
+            if path.is_file() && is_sway_file(&path) {
+                format_file(
+                    app,
+                    path,
+                    Some(workspace.dir().to_path_buf()),
+                    &mut formatter,
+                )?;
+            }
+        }
+    }
+
+    // Format subdirectories. Note that we do not call format on members directly here, since
+    // in workspaces, it is perfectly valid to have subdirectories containing Sway files,
+    // yet not be a member of the workspace.
+    for sub_dir in get_sway_dirs(dir.to_path_buf()) {
+        // Here, we cannot simply call Formatter::from_dir() and rely on defaults
+        // if there is not swayfmt.toml in the sub directory because we still want
+        // to use the swayfmt.toml at the workspace root (if any).
+        // In order of priority: member > workspace > default.
+        if members.contains(&sub_dir.join(constants::MANIFEST_FILE_NAME)) {
```

Comment:
Hmm, it also just occurred to me after your comment @sdankel that this doesn't handle the case where a subdir is not a member but has a `swayfmt.toml`. The [AMM](https://github.com/FuelLabs/sway-applications/tree/0132a4a89bb0f9c1a5d016db609adead87c43626/AMM) is one such example of a project - the forc projects within `libraries` are not declared as members.